File: saksi_mobile/lib/features/session/domain/entities/session.dart
enum Speaker { officer, customer, unknown }

enum SessionStatus { active, ended, aborted }

class Session {
  const Session({
    required this.id,
    required this.officerId,
    required this.productId,
    required this.status,
    required this.startedAt,
    this.score = 0,
  });

  final String id;
  final String officerId;
  final String productId;
  final SessionStatus status;
  final DateTime startedAt;
  final int score;

  bool get isRunning => status == SessionStatus.active;
}

class Utterance {
  const Utterance({
    required this.id,
    required this.speaker,
    required this.text,
    this.startMs = 0,
    this.revised = false,
  });

  final String id;
  final Speaker speaker;
  final String text;
  final int startMs;

  /// true kalau label pembicara pernah direvisi oleh diarization.
  final bool revised;

  Utterance copyWith({Speaker? speaker, bool? revised}) => Utterance(
        id: id,
        speaker: speaker ?? this.speaker,
        text: text,
        startMs: startMs,
        revised: revised ?? this.revised,
      );
}

File: saksi_frontend/src/domain/entities/session.ts
export type Speaker = 'officer' | 'customer' | 'unknown'
export type SessionStatus = 'active' | 'ended' | 'aborted'

export interface Session {
  id: string
  officerId: string
  productId: string
  status: SessionStatus
  startedAt: string
  score: number
}

export interface Utterance {
  id: string
  speaker: Speaker
  text: string
  startMs: number
  /** true kalau label speaker pernah direvisi oleh diarization */
  revised: boolean
  /** Label mentah diarization (mis. "A"/"B") — untuk verifikasi, bukan identitas. */
  sourceSpeaker?: string
}
